tmVhdlProducer/algodist.py

```diff
@@ -1694,6 +1697,16 @@ def dump_distribution(collection: ModuleCollection, filename: str):
 
 def distribute(eventSetup, modules: int, config: str, ratio: float, reverse_sorting: bool, constraints: Dict[str, str] = None) -> ModuleCollection:
     """Distribution wrapper function, provided for convenience."""
+
+    # hash value of the content of all .py and .vhd files in 'file_path'.
+    file_path = os.path.dirname(__file__)
+    files_hash_value = vhdlhelper.get_files_hash_value(file_path)
```

Comment:
Calculating the software hash is not part of the resource distribution, consider moving this into module `vhdlproducer` or `__main__`.

Reply:
The `constants` module is not ideal, I moved the code to the `__main__` module (entry point for hash calculation).